README
NAME

    Map::Metro::Plugin::Map::Helsinki - Map::Metro map for Helsinki

VERSION

    Version 0.1984, released 2015-01-16.

SYNOPSIS

        use Map::Metro;
        my $graph = Map::Metro->new('Helsinki')->parse;
    
        my $graph2 = Map::Metro->new('Helsinki', hooks => 'Helsinki::Swedish')->parse;
        # now the station names are in Swedish

DESCRIPTION

    See Map::Metro for usage information.

    This distribution also includes the
    Map::Metro::Plugin::Hook::Helsinki::Swedish hook, which if applied
    translates all station names into Swedish.

Status

    See Map::Metro::Plugin::Map::Helsinki::Lines

    This map includes:

      * The two branches of the Helsinki metro [wikipedia
      <https://en.wikipedia.org/wiki/Helsinki_Metro>]

SOURCE

    https://github.com/Csson/p5-Map-Metro-Helsinki

HOMEPAGE

    https://metacpan.org/release/Map-Metro-Plugin-Map-Helsinki

AUTHOR

    Erik Carlsson <info@code301.com>

COPYRIGHT AND LICENSE

    This software is copyright (c) 2015 by Erik Carlsson
    <info@code301.com>.

    This is free software; you can redistribute it and/or modify it under
    the same terms as the Perl 5 programming language system itself.



Changes
Revision history for Map-Metro-Plugin-Map-Helsinki

{{$NEXT}}
 [API Changes]

 [Bug Fixes]

 [Documentation]

 [Enhancements]

 [New Features]

 [Other Changes]

 [Requirements]

 [Stats & Dependency Changes]

 [Testing]

0.1987 2016-10-30T08:42:06Z UTC
 [Requirements]
 - No longer uses Moops in the Swedish hook.
 - Lowered Perl compat to 5.10.

 [Stats & Dependency Changes]
 - 5 files changed, 68 insertions(+), 68 deletions(-)
 - (run req) + Moose 2.1803
 - (run req) + namespace::autoclean 0.22
 - (run req) ~ Map::Metro 0.1900 --> 0.2405
 - (run req) ~ perl 5.016000 --> 5.010000
 - (run req) - Moose::Role
 - (test req) + IO::Handle (any)
 - (test req) + IPC::Open3 (any)
 - (conf req) + ExtUtils::MakeMaker (any)
 - (conf req) + File::ShareDir::Install 0.06
 - (conf req) - Module::Build::Tiny
 - (dev req) + Badge::Depot 0.0103
 - (dev req) + Badge::Depot::Plugin::Coverage (any)
 - (dev req) + Badge::Depot::Plugin::Cpantesters (any)
 - (dev req) + Badge::Depot::Plugin::Kwalitee (any)
 - (dev req) + Badge::Depot::Plugin::Perl (any)
 - (dev req) + Badge::Depot::Plugin::Travis (any)
 - (dev req) + Dist::Zilla 5.000
 - (dev req) + Dist::Zilla::Plugin::Authority 1.009
 - (dev req) + Dist::Zilla::Plugin::BumpVersionAfterRelease::Transitional
   0.007
 - (dev req) + Dist::Zilla::Plugin::ChangeStats::Dependencies::Git 0.0200
 - (dev req) + Dist::Zilla::Plugin::Clean 0.07
 - (dev req) + Dist::Zilla::Plugin::DistIller::MetaGeneratedBy (any)
 - (dev req) + Dist::Zilla::Plugin::Git 2.036
 - (dev req) + Dist::Zilla::Plugin::Git::CheckFor::CorrectBranch 0.013
 - (dev req) + Dist::Zilla::Plugin::Git::Commit (any)
 - (dev req) + Dist::Zilla::Plugin::Git::Contributors 0.017
 - (dev req) + Dist::Zilla::Plugin::InstallRelease 0.008
 - (dev req) + Dist::Zilla::Plugin::MakeMaker (any)
 - (dev req) + Dist::Zilla::Plugin::MetaConfig (any)
 - (dev req) + Dist::Zilla::Plugin::NextRelease::Grouped 0.0200
 - (dev req) + Dist::Zilla::Plugin::PromptIfStale (any)
 - (dev req) + Dist::Zilla::Plugin::RewriteVersion::Transitional 0.007
 - (dev req) + Dist::Zilla::Plugin::Test::Compile 2.054
 - (dev req) + Dist::Zilla::Plugin::Test::EOF 0.0501
 - (dev req) + Dist::Zilla::Plugin::Test::EOL 0.18
 - (dev req) + Dist::Zilla::Plugin::Test::Kwalitee::Extra v0.2.1
 - (dev req) + Dist::Zilla::Plugin::Test::NoTabs 0.15
 - (dev req) + Dist::Zilla::Plugin::Test::Version 1.05
 - (dev req) + Pod::Weaver::Section::GenerateSection 1.01
 - (dev req) + Test::EOF (any)
 - (dev req) + Test::EOL (any)
 - (dev req) + Test::More 0.88
 - (dev req) + Test::NoTabs 1.4
 - (dev req) + Test::Version 1
 - (dev req) ~ Dist::Zilla::Plugin::CheckChangesHasContent (any) --> 0.008
 - (dev req) ~ Dist::Zilla::Plugin::CopyFilesFromBuild (any) --> 0.151680
 - (dev req) ~ Dist::Zilla::Plugin::GithubMeta (any) --> 0.54
 - (dev req) ~ Dist::Zilla::Plugin::MetaProvides::Package (any) -->
   2.003001
 - (dev req) ~ Dist::Zilla::Plugin::PodWeaver (any) --> 4.006
 - (dev req) ~ Dist::Zilla::Plugin::ReadmeAnyFromPod (any) --> 0.150250
 - (dev req) ~ Dist::Zilla::Plugin::RunExtraTests (any) --> 0.028
 - (dev req) ~ Dist::Zilla::Plugin::Test::ReportPrereqs (any) --> 0.021
 - (dev req) ~ Pod::Weaver::Section::Badges (any) --> 0.0401
 - (dev req) ~ Test::Warnings (any) --> 0.026
 - (dev req) - Dist::Zilla::Plugin::CheckExtraTests
 - (dev req) - Dist::Zilla::Plugin::ContributorsFromGit
 - (dev req) - Dist::Zilla::Plugin::MetaProvides::Class
 - (dev req) - Dist::Zilla::Plugin::ModuleBuildTiny
 - (dev req) - Dist::Zilla::Plugin::NextRelease
 - (dev req) - Dist::Zilla::Plugin::NextVersion::Semantic
 - (dev req) - Dist::Zilla::Plugin::OurPkgVersion
 - (dev req) - Dist::Zilla::Plugin::PodnameFromClassname
 - (dev req) - Dist::Zilla::Plugin::Prereqs::Plugins
 - (dev req) - Dist::Zilla::Plugin::PreviousVersion::Changelog
 - (dev req) - Dist::Zilla::Plugin::ReversionOnRelease
 - (dev req) - Pod::Weaver::Section::Homepage::DefaultCPAN
 - (dev req) - Pod::Weaver::Section::Source::DefaultGitHub
 - (dev sug) + Dist::Iller 0.1408
 - (dev sug) + Dist::Iller::Config::Author::CSSON 0.0312

0.1985 2015-04-26T00:07:02Z CEST
 [Documentation]
 - README is now generated by dzil (kwalitee fix)

 [Meta]
 - Rebuilt from iller.yaml (changes to dist.ini, badges, and more).
 - No functional changes.

0.1984 2015-01-16T19:39:13Z CET
 [Enhancements]
 - Includes detailed line information
 - Includes visualization

 [Meta]
 - Added github repo to metadata
 - Added Moose as a prereq

0.1983 2014-12-13T21:41:18Z CET
 [Enhancements]
 - Test undiacriticed station names

0.1982 2014-12-12T23:55:59Z CET
 - Initial release


lib/Map/Metro/Plugin/Map/Helsinki.pm
use 5.10.0;
use strict;
use warnings;

package Map::Metro::Plugin::Map::Helsinki;

# ABSTRACT: Map::Metro map for Helsinki
# AUTHORITY
our $VERSION = '0.1988';

use Moose;
with 'Map::Metro::Plugin::Map';

has '+mapfile' => (
    default => 'map-helsinki.metro',
);
sub map_version {
    return $VERSION;
}
sub map_package {
    return __PACKAGE__;
}

1;

__END__

=encoding utf-8

=head1 SYNOPSIS

    use Map::Metro;
    my $graph = Map::Metro->new('Helsinki')->parse;

    my $graph2 = Map::Metro->new('Helsinki', hooks => 'Helsinki::Swedish')->parse;
    # now the station names are in Swedish

=head1 DESCRIPTION

See L<Map::Metro> for usage information.

This distribution also includes the C<Map::Metro::Plugin::Hook::Helsinki::Swedish> hook, which if applied
translates all station names into Swedish.

=head1 Status

See L<Map::Metro::Plugin::Map::Helsinki::Lines>

This map includes:

=for :list
* The two branches of the Helsinki metro [L<wikipedia|https://en.wikipedia.org/wiki/Helsinki_Metro>]

=for HTML <p><a href="https://raw.githubusercontent.com/Csson/p5-Map-Metro-Helsinki/master/static/images/helsinki.png"><img src="https://raw.githubusercontent.com/Csson/p5-Map-Metro-Helsinki/master/static/images/helsinki.png" style="max-width: 600px" /></a></p>

=cut


lib/Map/Metro/Plugin/Hook/Helsinki/Swedish.pm
use 5.10.0;
use strict;
use warnings;

package Map::Metro::Plugin::Hook::Helsinki::Swedish;

# ABSTRACT: Use the Swedish station names
# AUTHORITY
our $VERSION = '0.1988';

use Moose;
use namespace::autoclean;
use Map::Metro::Hook;
use Encode qw/encode_utf8 decode_utf8/;
use utf8;

sub register {
    before_add_station => sub {

        my $self = shift;
        my $station = shift;
        $station->set_name($self->station_translation($station->name));

    };
}

sub station_translation {
    my $self = shift;
    my $name = shift;

    my $to_swedish = {
        'Ruoholahti' => 'Gräsviken',
        'Kamppi' => 'Kampen',
        'Rautatientori' => 'Järnvägstorget',
        'Kaisaniemi' => 'Kajsaniemi',
        'Hakaniemi' => 'Hagnäs',
        'Sörnälnen' => 'Sörnäs',
        'Kalasatama' => 'Fiskhamnen',
        'Kulosaari' => 'Brändö',
        'Herttoniemi' => 'Hertonäs',
        'Siilitie' => 'Igelkottsvägen',
        'Itäkeskus' => 'Östra centrum',
        'Myllypuro' => 'Kvarnbäcken',
        'Kontula' => 'Gårdsbacka',
        'Mellunmäki' => 'Mellungsbacka',
        'Puotila' => 'Botby gård',
        'Rastila' => 'Rastböle',
        'Vuosaari' => 'Nordsjö',
    };
    return $to_swedish->{ $name } // $name;
}

1;


Makefile.PL
# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v6.007.
use strict;
use warnings;

use 5.010000;

use ExtUtils::MakeMaker;

use File::ShareDir::Install;
$File::ShareDir::Install::INCLUDE_DOTFILES = 1;
$File::ShareDir::Install::INCLUDE_DOTDIRS = 1;
install_share dist => "share";


my %WriteMakefileArgs = (
  "ABSTRACT" => "Map::Metro map for Helsinki",
  "AUTHOR" => "Erik Carlsson <info\@code301.com>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0,
    "File::ShareDir::Install" => "0.06"
  },
  "DISTNAME" => "Map-Metro-Plugin-Map-Helsinki",
  "LICENSE" => "perl",
  "MIN_PERL_VERSION" => "5.010000",
  "NAME" => "Map::Metro::Plugin::Map::Helsinki",
  "PREREQ_PM" => {
    "Map::Metro" => "0.2405",
    "Moose" => "2.1803",
    "namespace::autoclean" => "0.22"
  },
  "TEST_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0,
    "File::Spec" => 0,
    "IO::Handle" => 0,
    "IPC::Open3" => 0,
    "Test::More" => "0.96"
  },
  "VERSION" => "0.1988",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


my %FallbackPrereqs = (
  "ExtUtils::MakeMaker" => 0,
  "File::Spec" => 0,
  "IO::Handle" => 0,
  "IPC::Open3" => 0,
  "Map::Metro" => "0.2405",
  "Moose" => "2.1803",
  "Test::More" => "0.96",
  "namespace::autoclean" => "0.22"
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);

{
package
MY;
use File::ShareDir::Install qw(postamble);
}
